### PR TITLE
shortest path maintenance

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

i'm so bored it's not even funny

### CODE REVIEW

File: content/4_Gold/Shortest_Paths.mdx [hunk 1]
```diff
@@ -78,142 +78,172 @@ Floyd-Warshall is suitable because of the low $N (N \le 500)$, and the inclusion
...
-
-	vector<vector<ll>> mat(n, vector<ll>(n, INF));
+	std::cin >> n >> m >> q;
 
+	vector<vector<long long>> min_dist(n, vector<long long>(n, BIG));
 	for (int i = 0; i < m; i++) {
-		int u, v, w;
-		cin >> u >> v >> w, u--, v--;
-		if (w < mat[u][v]) mat[u][v] = mat[v][u] = w;
+		int a, b;
```
Comment (envyaims) [inline content/4_Gold/Shortest_Paths.mdx:99]: ?? why a b c?

File: content/4_Gold/Shortest_Paths.mdx [hunk 2]
```diff
@@ -78,142 +78,172 @@ Floyd-Warshall is suitable because of the low $N (N \le 500)$, and the inclusion
...
+		}
+	}
+
+	for (int i = 0; i < q; i++) {
+		int a, b;
+		std::cin >> a >> b;
+		a--;
+		b--;
+
+		// Handle some special cases
```
Comment (envyaims) [inline content/4_Gold/Shortest_Paths.mdx:125]: well what are they

**Review by envyaims — CHANGES_REQUESTED** [review]

File: content/4_Gold/Shortest_Paths.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply content/4_Gold/Shortest_Paths.mdx:99]: that’s what the problem uses

File: content/4_Gold/Shortest_Paths.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply content/4_Gold/Shortest_Paths.mdx:125]: they are literally right below

Comment (envyaims) [reply content/4_Gold/Shortest_Paths.mdx:125]: some explanation would be nice

File: content/4_Gold/Shortest_Paths.mdx (hunk 1 above)
Comment (envyaims) [reply content/4_Gold/Shortest_Paths.mdx:99]: doesnt mean its good practice

Comment (SansPapyrus683) [reply content/4_Gold/Shortest_Paths.mdx:99]: how come you’re fine with n and m but not this 💀

File: content/4_Gold/Shortest_Paths.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply content/4_Gold/Shortest_Paths.mdx:125]: alright, will do

File: content/4_Gold/Shortest_Paths.mdx (hunk 1 above)
Comment (envyaims) [reply content/4_Gold/Shortest_Paths.mdx:99]: find one person who unironically uses $a$ and $b$ for edges, and $c$ for weights

Comment (SansPapyrus683) [reply content/4_Gold/Shortest_Paths.mdx:99]: cases does apparently

Comment (ryanchou-dev) [reply content/4_Gold/Shortest_Paths.mdx:99]: it's fine as long as it's descriptive or it's from the problem

**Review by ryanchou-dev — APPROVED** [review]
just hear those sleigh bells jingling

ring ting tingling too

come on, it's lovely weather for a sleigh ride together with you

**Comment by SansPapyrus683** [comment]
IT ISN'T EVEN CHRISTMAS 😭

File: content/4_Gold/Shortest_Paths.mdx [hunk 3]
```diff
@@ -270,156 +300,151 @@ Used as the first step of the following:
...
-#define pb push_back
+int main() {
+	int n, m;
+	cin >> n >> m;
 
-vector<pair<int, int>> graph[100000];
-// Adjacency list of (neighbour, edge weight)
-ll dist[100000];
-int N;
+	// Adjacency list of (neighbour, edge weight)
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply content/4_Gold/Shortest_Paths.mdx:318]: i don't feel like this waffles the code too much
but if you insist, just tell me

[the remark this answers is not in the document]
Comment (ryanchou-dev) [reply content/4_Gold/Shortest_Paths.mdx:318]: okay makes sense

image

**Comment by ryanchou-dev** [comment]
it's not about christmas -- it's about having the ✨holiday spirit✨

**Pull request merged**